### PR TITLE
Add tech spec for Green Wrasse (GSI-79)

### PR DESCRIPTION
**Pull request by Cito**

### CODE REVIEW

File: 33-green-wrasse/technical_specification.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+Since the frontend first loads all relevant objects using this end point,
+these can be used in the detail view as well, and the backend does not need
+to provide a separate endpoint for getting individual Access Request Objects.
+
+Used by the web frontend to modify the status of an existing request:
+
+- `PATCH /access-requests/{access_request_id}`
+  - auth header: internal access token
+  - request body:
+    - `status`: enum (allowed/denied/pending)
```
Comment (KerstenBreuer) [inline 33-green-wrasse/technical_specification.md:127]: One could also model it as an optional boolean, where None means pending (no decision), `True` means approved, `False` means denied.

File: 33-green-wrasse/technical_specification.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+In order to know which data stewards should be notified, the configuration of
+the service should include a parameter that is a list of all email addresses
+that should be notified. This could be set to the email address of one or
+more responsible data steward(s), or just contain the email of the help-desk.
+
+The configuration should also contain parameters for the maximum time interval
+that the start date of the access request can be postponed and the maximum time
+interval between start date and end date (the maximum access validity period).
+This configuration must be added to both the Data Portal and the ARS, since it
+is used for default values and validation on both the frontend and the backend.
```
Comment (KerstenBreuer) [inline 33-green-wrasse/technical_specification.md:222]: If we don't allow changing the time right now, we might also skip the configuration of the Data Portal.

File: 33-green-wrasse/technical_specification.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+If the user specified in the access token does not have a data steward role,
+the `user_id` parameter is assumed to be the ID of that user,
+and an authorization error is returned if a different `user_id` is passed.
+
+The response body should be an array of Access objects as described above.
+
+For deleting (revoking) existing access grants in the Claims Repository,
+the already existing endpoint to delete a Claim object can be used, since the
+ID of the Data Access object is the same as the one of the corresponding
+Claim Object.
```
Comment (KerstenBreuer) [inline 33-green-wrasse/technical_specification.md:334]:
```suggestion
Claim Object.

Both endpoints should be proxied by the Access Request Service API to allow client access by data stewards.
```

**Review by KerstenBreuer — APPROVED** [review]
Looks excellent. Really nice description. Great drawings.

File: 33-green-wrasse/technical_specification.md (hunk 1 above)
Comment (Cito) [reply 33-green-wrasse/technical_specification.md:127]: Yes, but a boolean is more expressive, and we may want to add more status values later (like different stages of "pending" or "revoked" when it had been allowed and then later denied).

File: 33-green-wrasse/technical_specification.md (hunk 2 above)
Comment (Cito) [reply 33-green-wrasse/technical_specification.md:221]: Unless we want to show the default for the default access duration in the frontend.

File: 33-green-wrasse/technical_specification.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+### Configuration
+
+In order to know which data stewards should be notified, the configuration of
+the service should include a parameter that is a list of all email addresses
+that should be notified. This could be set to the email address of one or
+more responsible data steward(s), or just contain the email of the help-desk.
+
+The configuration should also contain parameters for the maximum time interval
+that the start date of the access request can be postponed and the maximum time
+interval between start date and end date (the maximum access validity period).
```
[the remark this answers is not in the document]
Comment (Cito) [reply 33-green-wrasse/technical_specification.md]: Mentioned that now.

File: 33-green-wrasse/technical_specification.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+The Access Request Service REST API should have the following endpoints:
+
+Used by the web frontend to create and view access requests:
+
+- `POST /access-requests`
+  - auth header: internal access token
+  - request body:
+    - `user_id`: string (the registered user ID of the requester)
+    - `dataset_id`: string (the ID of the requested dataset)
+    - `work_type`: enum (download or upload, must be download for now)
```
[the remark this answers is not in the document]
Comment (Cito) [reply 33-green-wrasse/technical_specification.md]: Ok, removed it.

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**